src/App.jsx

```diff
@@ -1,27 +1,108 @@
-import React, { useState, useEffect } from 'react';
-import './App.css';
-import TodoList from './TodoList';
-import AddTodoForm from './AddTodoForm';
+import React, { useState, useEffect } from "react";
+import "./App.css";
+import TodoList from "./TodoList";
+import AddTodoForm from "./AddTodoForm";
 
 function App() {
-  const [todoList, setTodoList] = useState([])
-  const [isLoading, setIsLoading] = useState(true)
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    const fetchData = () => {
-      return new Promise((resolve, reject) => {
-        setTimeout(() => {
-          const savedTodoList = localStorage.getItem("savedTodoList")
-          const initialTodoList = savedTodoList ? JSON.parse(savedTodoList) : []
-          resolve({ data: { todoList: initialTodoList } })
-        }, 2000)
-      });
+  //getting my tasks from airtable
+
+  const fetchData = async () => {
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
     };
+    try {
+      const response = await fetch(url, options);
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
 
-    fetchData().then(result => {
-      setTodoList(result.data.todoList)
+      const data = await response.json()
+
+      const todos = data.records.map((todo) => {
+        const newTodo = {
+          id: todo.id,
+          title: todo.fields.title,
+        };
+        return newTodo
+      });
+      setTodoList(todos)
       setIsLoading(false)
-    });
+    } catch (error) {
+      console.log(`Error: ${response.status}`)
+    }
+  };
+
+  //post task 
+  const postTodo = async (todo) => {
+    try {
+      const airtableData = {
+        fields: {
+          title: todo.title,
+        },
+      };
+      const response = await fetch(
+        `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${
+          import.meta.env.VITE_TABLE_NAME
+        }`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+          },
+          body: JSON.stringify(airtableData),
+        }
+      );
+      if (!response.ok) {
+        const message = `Error has occurred: ${response.status}`
+        throw new Error(`Error: ${response.status}`)
```

Comment:
The variable message has been declared but has never been used. You can use it when throwing a new Error or remove it.